### PR TITLE
Convert config unit tests

### PR DESCRIPTION
**Pull request by michaelgoin**
## Proposed Release Notes

* Converted config unit tests to fully use tap API and extracted related tests into more-specific test files.

## Details

Apologies for the PR size. On outset, I didn't realize how big the config.test.js file was or I would have approached slightly differently.

Converted all tests from the /config folder in unit tests.

The config.test.js file was massive and thus split out into more targeted test files, where seemed appropriate. This can help speed things up in some cases due to parallelization of test file runs. But mostly, I did this to ease finding of related tests and long-term maintenance.

Very little effort was put into validity of tests but rather focused on getting converted for consistency.

I did have to rebase onto the recent Node 16 / tap upgrade changes but didn't have much in the way of conflicts. Mostly just removing serverless_mode feature flag tests again when resolving conflicts. So hopefully didn't end up with any surprises.

### main

82675 passing (1m)
33 pending

### this change

82897 passing (1m)
33 pending

Additional assertions mostly due to how tap counts assertions but in some cases additional assertions were added or fixed assertions that were skipped accidentally in the mocha style.

### CODE REVIEW

**Review by bizob2828 — CHANGES_REQUESTED** [review]
just a few comments on replacing deprecated uses of
 * `t.deepEqual` w/ `t.same`
 * `t.notThrow` w/ `t.doesNotThrow`

File: test/unit/config/config-security.test.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+      policies[key].enabled = false
+    })
+
+    const response = config.applyLasp(agent, policies)
+    const payload = response.payload
+
+    t.equal(config.transaction_tracer.record_sql, 'off')
+    t.equal(agent._resetQueries.callCount, 1)
+    t.equal(config.attributes.include_enabled, false)
+    t.deepEqual(config.attributes.exclude, ['request.parameters.*'])
```
[the remark this answers is not in the document]
Comment (michaelgoin) [reply test/unit/config/config-security.test.js:167]: ah yeah, was converted pre-tap migration but i can update.

File: test/unit/config/collector_hostname.test.js [hunk 2]
```diff
@@ -5,29 +5,55 @@
 
 'use strict'
 
-// TODO: convert to normal tap style.
-// Below allows use of mocha DSL with tap runner.
-require('tap').mochaGlobals()
+const tap = require('tap')
```
[the remark this answers is not in the document]
Comment (michaelgoin) [reply test/unit/config/collector_hostname.test.js:8]: yeah, i can clean that up.

**Comment by michaelgoin** [comment]
OK, should have all of the deprecation warnings resolved and renamed the file.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**